egs/wsj/s5/utils/parallel/slurm.pl

```diff
@@ -1,628 +1,391 @@
 #!/usr/bin/env perl
-use strict;
-use warnings;
 
+# SPDX-License-Identifier: Apache-2.0
 # Copyright 2012  Johns Hopkins University (Author: Daniel Povey).
 #           2014  Vimal Manohar (Johns Hopkins University)
 #           2015  Johns Hopkins University (Yenda Trmal <jtrmal@gmail.com>>)
-# Apache 2.0.
+#           2020  Kirill 'kkm' Katsnelson <kkm@pobox.com>
+
+use strict;
+use warnings;
 
 use File::Basename;
-use Cwd;
-use Getopt::Long;
-
-# slurm.pl was created from the queue.pl
-# queue.pl has the same functionality as run.pl, except that
-# it runs the job in question on the queue (Sun GridEngine).
-# This version of queue.pl uses the task array functionality
-# of the grid engine.  Note: it's different from the queue.pl
-# in the s4 and earlier scripts.
-
-# The script now supports configuring the queue system using a config file
-# (default in conf/queue.conf; but can be passed specified with --config option)
-# and a set of command line options.
-# The current script handles:
-# 1) Normal configuration arguments
-# For e.g. a command line option of "--gpu 1" could be converted into the option
-# "-q g.q -l gpu=1" to qsub. How the CLI option is handled is determined by a
-# line in the config file like
-# gpu=* -q g.q -l gpu=$0
-# $0 here in the line is replaced with the argument read from the CLI and the
-# resulting string is passed to qsub.
-# 2) Special arguments to options such as
-# gpu=0
-# If --gpu 0 is given in the command line, then no special "-q" is given.
-# 3) Default argument
-# default gpu=0
-# If --gpu option is not passed in the command line, then the script behaves as
-# if --gpu 0 was passed since 0 is specified as the default argument for that
-# option
-# 4) Arbitrary options and arguments.
-# Any command line option starting with '--' and its argument would be handled
-# as long as its defined in the config file.
-# 5) Default behavior
-# If the config file that is passed using is not readable, then the script
-# behaves as if the queue has the following config file:
-# $ cat conf/queue.conf
-# # Default configuration
-# command sbatch --export=PATH  -S /bin/bash -j y -l arch=*64*
-# option mem=* --mem-per-cpu $0
-# option mem=0          # Do not add anything to qsub_opts
-# option num_threads=* --cpus-per-task $0
-# option num_threads=1  # Do not add anything to qsub_opts
-# option max_jobs_run=* -tc $0
-# default gpu=0
-# option gpu=0 -p shared
-# option gpu=*  -p gpu  #this has to be figured out
-
-#print STDERR "$0 " . join(" ", @ARGV) . "\n";
-
-my $qsub_opts = "";
-my $sync = 0;
-my $num_threads = 1;
-my $max_jobs_run;
-my $gpu = 0;
-
-my $config = "conf/slurm.conf";
```

Comment:
see this....

Reply:
Will fix, thanks for spotting! What's your opinion on the default hardcoded configuration in the file: should we keep it, or drop and always require a config file?

My take on this is that no hardcoded defaults could be in any way sensible. Tuning a distributed configuration is always required, in my experience. I'd rather ship a default config file. IMO, the mere act of copying the file into conf/ will draw attention to it. I'll also comment it dearly.

Also, I want to make a pinned post to kaldi-help a few days before merging. Like, nearly right now. Stuff can break, and we don't know how many Slurm users are out there.